File: tests/test-ggwave.py
import sys
import ggwave

# optionally disable logging
#ggwave.disableLog()

# create ggwave instance with default parameters
instance = ggwave.init()

payload = 'hello python'

# generate audio waveform for string "hello python"
waveform = ggwave.encode(payload, protocolId = 1, volume = 20, instance = instance)

# decode the audio waveform back to text
res = ggwave.decode(instance, waveform)

if res != payload.encode():
    sys.exit(1)

# disable the Rx protocol - the decoding should fail
ggwave.rxToggleProtocol(protocolId = 1, state = 0)
instanceTmp = ggwave.init()

res = ggwave.decode(instanceTmp, waveform)
if res != None:
    sys.exit(1)

ggwave.free(instanceTmp);

# re-enable the Rx protocol - the decoding should succeed
ggwave.rxToggleProtocol( protocolId = 1, state = 1)
instanceTmp = ggwave.init()

res = ggwave.decode(instance, waveform)
if res != payload.encode():
    sys.exit(1)

ggwave.free(instanceTmp);

ggwave.free(instance);

File: tests/test-ggwave.js
var factory = require('../bindings/javascript/ggwave.js')

factory().then(function(ggwave) {
    // create ggwave instance with default parameters
    var parameters = ggwave.getDefaultParameters();

    parameters.operatingMode |= ggwave.GGWAVE_OPERATING_MODE_USE_DSS;

    var instance = ggwave.init(parameters);
    console.log('instance: ' + instance);

    var payload = 'hello js';

    // generate audio waveform for string "hello js"
    var waveform = ggwave.encode(instance, payload, ggwave.ProtocolId.GGWAVE_PROTOCOL_AUDIBLE_FAST, 10);

    // decode the audio waveform back to text
    var res = ggwave.decode(instance, waveform);

    if (new TextDecoder("utf-8").decode(res) != payload) {
        process.exit(1);
    }
});
